orchestrator/db/filters/filters.py

```diff
@@ -32,57 +34,103 @@ class Filter(BaseModel):
 
 QueryType = Select
 ValidFilterFunctionsByColumnType = dict[str, Callable[[QueryType, str], QueryType]]
+ColumnMappings = dict[str, WhereCondGenerator]
 
 
 def generic_filters_validate(
-    valid_filter_functions_by_column: ValidFilterFunctionsByColumnType,
+    valid_columns: Iterable[str],
 ) -> Callable[[list[Filter]], tuple[Iterable[Filter], Iterable[Filter]]]:
     def _validate_filters(filter_by: list[Filter]) -> tuple[Iterable[Filter], Iterable[Filter]]:
         def _is_valid_filter(item: Filter) -> bool:
-            return item.field in valid_filter_functions_by_column and item.value is not None
+            return camel_to_snake(item.field) in valid_columns and item.value is not None
 
         return partition(_is_valid_filter, filter_by)
 
     return _validate_filters
 
 
-def generic_apply_filters(
-    valid_filter_functions_by_column: ValidFilterFunctionsByColumnType,
+_re_split = re.compile("[-|]")
+
+
+def _filter_to_node(filter_item: Filter) -> Node:
+    value = filter_item.value
+    if value and value[0] == "!":
+        updated_filter_item = Filter(field=filter_item.field, value=value[1:])
+        return "Negation", _filter_to_node(updated_filter_item)
+
+    # Workaround to deal with date fields. These should not be split like other fields.
+    # Fix after deprecating '-' as split-separator in favor of '|'
+    values = _re_split.split(filter_item.value) if "date" not in filter_item.field.lower() else [filter_item.value]
+    key_node = "Word", filter_item.field
+    value_node: Node
+    if len(values) > 1:
+        value_node = "ValueGroup", [("Word", v) for v in values if v]
+    else:
+        value_node = "Word", value
+
+    if len(value_node[1]) == 0:
+        raise Exception("Invalid filter arguments")
+
+    return "KVTerm", (key_node, value_node)
```

Comment:
When the terms `ValueGroup` and `KVTerm` were only used in the `utils/search_query.py` module it was manageable, but now that we're using them in other modules I think they should become constant/enum variables to avoid typo's. Also the devs who don't know about the search query stuff can then navigate to the definition for context